CRM/Utils/Token.php

```diff
@@ -1700,6 +1718,93 @@ public static function replaceMultipleContributionTokens($separator, $str, &$con
   }
 
   /**
+   * Get replacement strings for any event tokens
+   * @param string $token
+   * @param array $event an api result array for a single event
+   * @param bool $escapeSmarty
+   * @return string token replacement
+   */
+  public static function getEventTokenReplacement($token, $event, $escapeSmarty = FALSE) {
+    $entity = 'event';
+    self::_buildEventTokens();
+
+    $params = array('entity_id' => $event['id'], 'entity_table' => 'civicrm_event');
+
+    switch ($token) {
+      case 'balance':
```

Comment:
What does "balance" means? Is it outstanding participant payment. If yes, then instead of $params['entity_id']($event['id']) participant_id should be used to get getPaymentInfo.
Currently it displays $0.00 for the participant having balance